src/devtools/tests/devtools.test.tsx

```diff
@@ -0,0 +1,321 @@
+import React from 'react'
+import {
+  fireEvent,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+} from '@testing-library/react'
+import { QueryClient, QueryCache, useQuery } from '../..'
+import { getByTextContent, renderWithClient, sleep } from './utils'
+
+describe('ReactQueryDevtools', () => {
+  const queryCache = new QueryCache()
+  const queryClient = new QueryClient({
+    queryCache,
+    defaultOptions: {
+      queries: {
+        staleTime: 0,
+      },
+    },
+  })
+
+  beforeEach(() => {
+    queryCache.clear()
+  })
```

Comment:
I would prefer if we'd either:
- create a new QueryClient for each test so that they can run in isolation
- use the same cache and separate concerns by using unique query keys (there is a `queryKey()` util function that we use in other tests to ensure this.

That way, tests are more isolated and we can also run them potentially in parallel.

Reply:
Got it using the former method here because we rely on only a certain number of items being visible in devtools for some tests, hence creating a fresh client and cache for every test via a helper